compiler/transforms/reuse_memref_allocs.py

```diff
@@ -237,7 +255,10 @@ def get_new_memref_op(
                 # This happens when the dim is called on an input argument
                 return memref.Dim.from_source_and_index(memref_ssa, index_constant)
             if isinstance(memref_op, memref.Subview):
-                new_op = memref_op.sizes[index].owner
+                subview_size = get_subview_dim(memref_op, index)
+                if isinstance(subview_size, int):
+                    return arith.Constant.from_int_and_width(subview_size, IndexType())
```

Comment:
Do you need to know the size itself to perform this pass? Because otherwise you could probably use a memref.dim op, no?